pkg/envoy/route/route_config.go

```diff
@@ -52,6 +52,9 @@ const (
 
 	// httpHostHeader is the name of the HTTP host header
 	httpHostHeader = "host"
```

Comment:
let's rename this to `hostHeaderKey` and clarify in the comment that this key is used to specify the HTTP Host header in the `HTTPRouteMatch.Headers`